src/components/ToDo.tsx

```diff
@@ -0,0 +1,120 @@
+import * as React from 'react';
+import { useState } from 'react';
+import EditIcon from '@mui/icons-material/Edit';
+import DeleteIcon from '@mui/icons-material/Delete';
+import {
+    Typography,
+    TextField,
+    Divider,
+    Button,
+    Box,
+    List,
+    ListItem,
+    ListItemText,
+    ListItemSecondaryAction
+} from '@mui/material';
+
+interface TodoItem {
+    id: number;
+    text: string;
+}
+
+
+
+const ToDo = () => {
+    const [inputText, setInputText] = useState('');
+    const [todos, setTodos] = useState<TodoItem[]>([]);
+    const [editTodoId, setEditTodoId] = useState<number | null>(null);
+
+    const handleAddTodo = () => {
+        if (inputText) {
+            if (editTodoId !== null) {
+                const updatedTodos = todos.map(todo => {
+                    if (todo.id === editTodoId) {
+                        return { ...todo, text: inputText };
+                    }
+                    return todo;
+                });
+                setTodos(updatedTodos);
+                setEditTodoId(null);
+            } else {
+                const newTodo: TodoItem =
+                {
+                    id: Date.now(),
+                    text: inputText,
+                };
+                setTodos([...todos, newTodo]);
+            }
+            setInputText('');
+        }
+    };
+
+    const handleEditTodo = (id: number) => {
+        const todoToEdit = todos.find(todo => todo.id === id);
+        if (todoToEdit) {
+            setInputText(todoToEdit.text);
+            setEditTodoId(id);
+        }
+    };
+
+    const handleDeleteTodo = (id: number) => {
+        const updatedTodos = todos.filter(todo => todo.id !== id);
```

Comment:
I guess we should not be removing todo when deleting. It should just cross the text

![Uploading image.png…]()

Reply:
ok i will implement it